src/main/java/zowe/client/sdk/zosjobs/SubmitJobs.java

```diff
@@ -115,7 +117,10 @@ public Job submitJclCommon(SubmitJclParams params) throws Exception {
             headers.put(key, value);
         }
         if (params.getJclSymbols().isPresent()) {
-            // TODO..
+            Map<String, String> extraHeaders = getSubstitutionHeaders(params.getJclSymbols().get());
+            for (Map.Entry<String, String> entry : extraHeaders.entrySet()) {
+                headers.put(entry.getKey(), entry.getValue());
+            }
         }
```

Comment:
you may be able to replace this entire for statement with the following if you want and if it works:
extraHeaders.forEach((k, v) -> headers.put(k, v));

Reply:
you may even be able to replace the entire if and for statements with the following:
params.getJclSymbols().ifPresent(syms - > syms.forEach((k, v) -> headers.put(k, v)));